### PR TITLE
[Magiclysm] Add feral stormshapers

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Magiclysm] Add feral stormshapers"

#### Purpose of change

Still adding more feral mages

#### Describe the solution

Add the feral stormshaper and feral high stormshaper. They blast lightning around (of course), stun you with Wall of Fog, and cast Call Stormhammer and proceed to hammer the storms out of you. If you know Dispel Magic, you can dispel the stormhammer.

Also, the high stormshaper brings back Shockstorm (if anyone should have it, a lightning wizard should)

#### Testing

Feral stormshapers summon the stormhammer, which is dispellable. They use their spells appropriately (for monster AI).  Don't end up in melee and get downed by Wall of Fog while they have their stormhammer out. 

As with most of these feral wizards, they have no good defense against Bullet.

### CODE REVIEW

**Comment by Psithief** [comment]
So are they _more_ annoying than an electric zombie? They'll break all my electronics as well?

File: data/mods/Magiclysm/monsters/feral_wizards.json [hunk 1]
```diff
@@ -772,5 +772,219 @@
...
+        "type": "spell",
+        "spell_data": { "id": "feral_stormshaper_summon_stormhammer_spell" },
+        "cooldown": 50,
+        "condition": { "and": [ { "not": { "u_has_effect": "effect_feral_stormshaper_stormshammer" } }, { "not": { "u_has_flag": "NO_SPELLCASTING" } } ] },
+        "monster_message": "%1$s snaps an arm out and a hammer made of lightning crackles into existence."
+      },
+      {
+        "type": "monster_attack",
+        "attack_type": "melee",
+        "id": "feral_stormshaper_spell_lightning_blast",
```
Comment (rty275) [inline data/mods/Magiclysm/monsters/feral_wizards.json:861]: I think the ID on this needs to change to something like "feral_stormshaper_stormhammer_attack" or something like that? I think you already used "feral_stormshaper_spell_lightning_blast" above.

File: data/mods/Magiclysm/monsters/feral_wizards.json [hunk 2]
```diff
@@ -772,5 +772,219 @@
...
+        "type": "spell",
+        "spell_data": { "id": "feral_stormshaper_summon_stormhammer_spell" },
+        "cooldown": 40,
+        "condition": { "and": [ { "not": { "u_has_effect": "effect_feral_stormshaper_stormshammer" } }, { "not": { "u_has_flag": "NO_SPELLCASTING" } } ] },
+        "monster_message": "%1$s snaps an arm out and a hammer made of lightning crackles into existence."
+      },
+      {
+        "type": "monster_attack",
+        "attack_type": "melee",
+        "id": "feral_stormshaper_spell_lightning_blast",
```
Comment (rty275) [inline data/mods/Magiclysm/monsters/feral_wizards.json:963]: Same thing with the ID here, I think.

File: data/mods/Magiclysm/effects/effects.json [hunk 3]
```diff
@@ -1452,6 +1452,13 @@
     "rating": "good",
     "base_mods": { "hit_mod": [ 2 ] }
   },
+  {
+    "type": "effect_type",
+    "id": "effect_feral_stormshaper_stormshammer",
```
Comment (rty275) [inline data/mods/Magiclysm/effects/effects.json:1457]: I think you have an extra s in the "stormhammer" here? Unless it's supposed to be storm**s**hammer. Would also mean changing all the IDs in the attacks of course.

**Review by rty275 — COMMENTED** [review]
Hope I'm not overstepping by reviewing out of nowhere, I just like keeping up-to-date on mods and I've got enough json know-how to maybe help a bit?

**Comment by Standing-Storm** [comment]
> So are they _more_ annoying than an electric zombie? They'll break all my electronics as well?

Stormshapers don't have an EMP spell, so no. It's currently not possible to use magic to create EMPs.

But since they have other magic they'll definitely be more annoying than an electric zombie.

File: data/mods/Magiclysm/effects/effects.json (hunk 3 above)
Comment (Standing-Storm) [reply data/mods/Magiclysm/effects/effects.json:1457]: In this instance, the effect is only referenced by the monsters in attack conditions and isn't ever visible to the player, so as long as the id is consistent the exact id isn't important.

Comment (rty275) [reply data/mods/Magiclysm/effects/effects.json:1457]: Ahh, gotcha! I'll keep that in mind for the future.

**Comment by rty275** [comment]
Oh gosh, I just realized I reviewed while it was still in draft, you were probably going to catch that stuff yourself. I'm sorry, that was wrong of me.

**Comment by Standing-Storm** [comment]
No offense taken! Reviewing is mostly thankless work and I won't complain about someone willing to do it.

**Comment by Psithief** [comment]
> But if stormshaping does get an EMP spell, then why wouldn't feral stormshapers be able to cast it?

Feral monsters should be too smart to waste their time using an EMP spell when they can use a direct damage spell. An EMP wouldn't improve their chances to kill in almost all circumstances. Adding one would just be player punishment. Even the incandescent husk isn't firing off its EMP on purpose as far as a player is aware.

File: data/mods/Magiclysm/monsters/feral_wizards.json [hunk 4]
```diff
@@ -772,5 +772,229 @@
...
+        "accuracy": 4,
+        "damage_max_instance": [ { "damage_type": "electric", "amount": 20 } ],
+        "dodgeable": true,
+        "blockable": false,
+        "condition": { "not": { "u_has_flag": "NO_SPELLCASTING" } },
+        "hit_dmg_u": "%1$s gestures and sparks impact your %2$s!",
+        "hit_dmg_npc": "%1$s gestures and sparks impact <npcname>'s %2$s!",
+        "miss_msg_u": "%1$s gestures and you narrowly avoid a shower of sparks!",
+        "miss_msg_npc": "%1$s gestures and <npcname> narrowly avoids a shower of sparks!",
+        "no_dmg_msg_u": "%1$s gestures your %2$s is impacted by a shower of sparks bolt, but they do no damage.",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/monsters/feral_wizards.json:823]:
```suggestion
        "no_dmg_msg_u": "%1$s gestures and your %2$s is impacted by a shower of sparks bolt, but they do no damage.",
```

File: data/mods/Magiclysm/monsters/feral_wizards.json [hunk 5]
```diff
@@ -772,5 +772,229 @@
...
+        "accuracy": 6,
+        "damage_max_instance": [ { "damage_type": "electric", "amount": 25 } ],
+        "dodgeable": true,
+        "blockable": false,
+        "condition": { "not": { "u_has_flag": "NO_SPELLCASTING" } },
+        "hit_dmg_u": "%1$s gestures and sparks impact your %2$s!",
+        "hit_dmg_npc": "%1$s gestures and sparks impact <npcname>'s %2$s!",
+        "miss_msg_u": "%1$s gestures and you narrowly avoid a shower of sparks!",
+        "miss_msg_npc": "%1$s gestures and <npcname> narrowly avoids a shower of sparks!",
+        "no_dmg_msg_u": "%1$s gestures your %2$s is impacted by a shower of sparks bolt, but they do no damage.",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/monsters/feral_wizards.json:937]:
```suggestion
        "no_dmg_msg_u": "%1$s gestures and your %2$s is impacted by a shower of sparks bolt, but they do no damage.",
```

**Comment by Standing-Storm** [comment]
> Feral monsters should be too smart to waste their time using an EMP spell when they can use a direct damage spell. An EMP wouldn't improve their chances to kill in almost all circumstances. Adding one would just be player punishment. Even the incandescent husk isn't firing off its EMP on purpose as far as a player is aware.

That is more tactical acumen than the feral mages have. That's why the feral magus will put multiple debuffs on you and then cast dispel magic (removing all the debuffs they put on you). A feral stormshaper with an EMP spell would just be firing it off at an enemy, especially considering any such spell would probably do lightning damage _and_ EMP, rather than just being an EMP.  If you put one in combat with a mi-go they'd try to EMP it too.

**Comment by Maleclypse** [comment]
Do you want some feral amateur spellcaster variants?  Like one that randomly casts faerie fire or whatever the equivalent Magiclysm spell is on enemies at night time?

**Comment by Standing-Storm** [comment]
> Do you want some feral amateur spellcaster variants? Like one that randomly casts faerie fire or whatever the equivalent Magiclysm spell is on enemies at night time?

I actually put Faerie Fire into Magiclysm as a Magus spell! #76240 

If there's some way to give random ferals a chance to have a single spell that would be amazing--Magiclysm lore is that ~25% of the population could cast at least 1 spell so a huge portion of ferals should know magic (though only 10% of those people knew more than 1, so the proportion of feral mages to other ferals is probably right).

**Comment by Maleclypse** [comment]
So it would require doing the same shenanigans as the ratkin, where you have a new mon_id that is identical to the pre-existing except the small change and then drop that into spawngroups at an appropriate ratio.

**Pull request merged**